### ISSUE TITLE
Rewrite GitHub links to exact commit, not latest master

### ISSUE THREAD

**Issue by felixfbecker** [Microsoft/vscode-vsce#85]
I recently [added a new demo GIF to my debug adapter](https://github.com/felixfbecker/vscode-php-debug/commit/6374af2f24e464d7170eb1de7d920c0c76104301), without publishing a new release. I was surprised to see the new GIF already showing up in the [VS Code marketplace](https://marketplace.visualstudio.com/items?itemName=felixfbecker.php-debug), even though it is still the old README. Then I looked inside the vsix file and saw that vsce changed the relative link to the GIF to
```md
![Demo GIF](https://github.com/felixfbecker/vscode-php-debug/raw/master/images/demo.gif)
```
which means the demo GIF will always use the latest master. If I would make a commit that deletes the image it would completely break the marketplace appearance with a 404. The GIF is actually showing new features that are not available in the version in the marketplace, so it is misleading aswell...

Instead, the link should be rewritten to `https://raw.githubusercontent.com/felixfbecker/vscode-php-debug/137d4d6ca58393b05bac6dc41a54a8c4f1b1c6c1/images/demo.gif` (exact commit)

**Comment by kumarharsh** [comment]
@joaomoreno would change [these lines](https://github.com/Microsoft/vscode-vsce/blob/10ca06c4329bbe5e35163fc80df8455bc2b62391/src/package.ts#L311) to have current commit sha be enough to fix the links?

**Comment by joaomoreno** [comment]
I guess it would yes!

**Comment by joaomoreno** [comment]
@felixfbecker This is shaky ground. It would involve not allowing users to run `vsce publish` if their repository has anything dirty, since vsce wouldn't be able to know which would be the commit ID of the current changes.

**Comment by felixfbecker** [comment]
Correct. That is btw the way `npm version` behaves too. You cannot run it when your working directory is dirty.

**Comment by joaomoreno** [comment]
Let's do it then!

**Comment by pschaeflein** [comment]
vsce should not be defaulting to a branch named "master".

I eventually found the `--githubBranch` flag, which is not documented.
